protocols/request-response/tests/cbor.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! Integration tests for the `cbor::Behaviour`.
+
+use libp2p_request_response as request_response;
+use libp2p_request_response::cbor::Behaviour;
+use libp2p_request_response::{Config, ProtocolSupport};
+use libp2p_swarm::{StreamProtocol, Swarm};
+use libp2p_swarm_test::SwarmExt;
+use rand::{self, Rng};
+use serde::{Deserialize, Serialize};
+use std::iter;
+
+#[async_std::test]
+async fn cbor() {
```

Comment:
Can we replace this with a unit-test of the `Codec` implementation? `request-response` is already tested so this seems redundant.

You can use https://docs.rs/futures_ringbuf/0.3.1/futures_ringbuf/struct.Endpoint.html to construct a pair of `AsyncRead + AsyncWrite`.

Reply:
Of course, we can test only the `Codec` implementation. But we additionally have the `Behaviour` implementation, which in that case would not be covered by tests.

Reply:
We are just delegating through there. If that compiles without warnings, I am confident it is bug free :)